src/cpp/session/SessionMain.cpp

```diff
@@ -2976,10 +2990,13 @@ int main (int argc, char * const argv[])
       if (error)
          return sessionExitFailure(error, ERROR_LOCATION);
       
-      // start http connection listener
-      error = startHttpConnectionListener();
-      if (error)
-         return sessionExitFailure(error, ERROR_LOCATION);
+      // start http connection listener (TODO: not if running tests)
```

Comment:
Why not start the connection listener when we are running tests? (might as well test the fact that it can initialize, bind to it's port, etc.?)